### PR TITLE
Fix warnings for rust 1.80

### PR DESCRIPTION
**Pull request by gui1117**
Fix warnings for rust 1.80

### CODE REVIEW

File: substrate/primitives/runtime-interface/src/lib.rs [hunk 1]
```diff
@@ -17,7 +17,6 @@
 
 // Custom inner attributes are unstable, so we need to faky disable the attribute.
 // rustfmt still honors the attribute to not format the rustdocs below.
-#![cfg_attr(feature = "never", rustfmt::skip)]
```
Comment (bkchr) [inline substrate/primitives/runtime-interface/src/lib.rs:20]: Did you read the comment above? :D Not sure this is already fixed in rustfmt?

File: substrate/primitives/weights/Cargo.toml [hunk 2]
```diff
@@ -35,6 +35,7 @@ std = [
 	"sp-arithmetic/std",
 	"sp-debug-derive/std",
 ]
+runtime-benchmarks = []
```
Comment (bkchr) [inline substrate/primitives/weights/Cargo.toml:38]: Is there code in `sp-weights` that is guarded by the feature?

File: Cargo.toml [hunk 3]
```diff
@@ -542,6 +542,7 @@ default-members = [
 
 [workspace.lints.rust]
 suspicious_double_ref_op = { level = "allow", priority = 2 }
+unexpected_cfgs = { level = "allow", check-cfg = ['cfg(substrate_runtime)'] }
```
Comment (bkchr) [inline Cargo.toml:545]: What does this do?

File: substrate/primitives/weights/Cargo.toml (hunk 2 above)
Comment (gui1117) [reply substrate/primitives/weights/Cargo.toml:38]: yes this, we could also deprecate it IMO:
https://github.com/paritytech/polkadot-sdk/blob/18db502172bdf438f086cd5964c646b318b8ad37/substrate/primitives/weights/src/weight_v2.rs#L404-L416

File: substrate/primitives/runtime-interface/src/lib.rs (hunk 1 above)
Comment (gui1117) [reply substrate/primitives/runtime-interface/src/lib.rs:20]: sorry I missed it, but I did run `cargo fmt -p sp-runtime-interface` the file was unchanged so I thought it wasn't relevant anymore.

File: Cargo.toml (hunk 3 above)
Comment (gui1117) [reply Cargo.toml:545]: it allows crate to use the condition `substrate_runtime` in `cfg` without warning like this:

```
warning: unexpected `cfg` condition name: `substrate_runtime`
   --> substrate/primitives/runtime-interface/src/lib.rs:378:66
    |
378 | #[cfg(all(any(target_arch = "riscv32", target_arch = "riscv64"), substrate_runtime))]
    |                                                                  ^^^^^^^^^^^^^^^^^
    |
    = help: expected names are: `clippy`, `debug_assertions`, `doc`, `docsrs`, `doctest`, `feature`, `miri`, `overflow_checks`, `panic`, `proc_macro`, `relocation_model`, `rustfmt`, `sanitize`, `sanitizer_cfi_generalize_pointers`, `sanitizer_cfi_normalize_integers`, `target_abi`, `target_arch`, `target_endian`, `target_env`, `target_family`, `target_feature`, `target_has_atomic`, `target_has_atomic_equal_alignment`, `target_has_atomic_load_store`, `target_os`, `target_pointer_width`, `target_thread_local`, `target_vendor`, `test`, `ub_checks`, `unix`, and `windows`
    = help: consider using a Cargo feature instead
    = help: or consider adding in `Cargo.toml` the `check-cfg` lint config for the lint:
             [lints.rust]
             unexpected_cfgs = { level = "warn", check-cfg = ['cfg(substrate_runtime)'] }
    = help: or consider adding `println!("cargo::rustc-check-cfg=cfg(substrate_runtime)");` to the top of the `build.rs`
    = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
    = note: `#[warn(unexpected_cfgs)]` on by default

```

File: substrate/frame/support/src/lib.rs [hunk 4]
```diff
@@ -869,7 +869,7 @@ macro_rules! hypothetically_ok {
 pub use serde::{Deserialize, Serialize};
 
 #[doc(hidden)]
-#[cfg(not(no_std))]
+#[cfg(feature = "std")]
```
Comment (ggwpez) [inline substrate/frame/support/src/lib.rs:872]:
```suggestion
```

I think this was a no-op previously and causes errors if you guard it now. CI should tell.

File: substrate/primitives/runtime-interface/src/lib.rs (hunk 1 above)
Comment (ggwpez) [reply substrate/primitives/runtime-interface/src/lib.rs:20]: The CI uses a specific nightly for formatting: `cargo +nightly-2024-04-14 fmt`.

File: substrate/frame/support/src/lib.rs (hunk 4 above)
Comment (gui1117) [reply substrate/frame/support/src/lib.rs:872]: but `no_std` is defined in frame-support with this line:
```
#![cfg_attr(not(feature = "std"), no_std)]
```

File: substrate/primitives/runtime-interface/src/lib.rs (hunk 1 above)
Comment (gui1117) [reply substrate/primitives/runtime-interface/src/lib.rs:20]: Also tested with this version, anyway CI should tell if the formatting of the crate is changing with this PR, in this case I'll revert this change.

File: substrate/frame/support/src/lib.rs [hunk 5]
```diff
@@ -869,7 +869,6 @@ macro_rules! hypothetically_ok {
 pub use serde::{Deserialize, Serialize};
 
 #[doc(hidden)]
-#[cfg(not(no_std))]
```
Comment (gui1117) [inline substrate/frame/support/src/lib.rs:872]: This actually needed in both std and not std: https://github.com/paritytech/polkadot-sdk/pull/5150#discussion_r1692787358

File: substrate/primitives/weights/src/weight_v2.rs [hunk 6]
```diff
@@ -401,14 +401,14 @@ where
 	}
 }
 
-#[cfg(any(test, feature = "std", feature = "runtime-benchmarks"))]
```
Comment (gui1117) [inline substrate/primitives/weights/src/weight_v2.rs:404]: Better not to add some conditional code for runtime-benchmark here IMO.
This code was never generated, better not to start generating it.

File: substrate/frame/support/src/lib.rs (hunk 4 above)
Comment (ggwpez) [reply substrate/frame/support/src/lib.rs]: Yea its unfortunate, as we *should not* need it in WASM, but no clue why it gets pulled in 🤷‍♂️

File: substrate/primitives/weights/src/weight_v2.rs (hunk 6 above)
Comment (ggwpez) [reply substrate/primitives/weights/src/weight_v2.rs:404]: Yea feel free to delete. I normally generate [helpers](https://github.com/paritytech/polkadot-sdk/blob/7ecf3f757a5d6f622309cea7f788e8a547a5dce8/substrate/frame/message-queue/src/mock_helpers.rs#L34) on the fly if i need to.

File: substrate/primitives/weights/Cargo.toml (hunk 2 above)
Comment (bkchr) [reply substrate/primitives/weights/Cargo.toml]: If it was never used, let's remove it.

File: Cargo.toml (hunk 3 above)
Comment (bkchr) [reply Cargo.toml:546]: CLippy...

File: substrate/frame/support/src/lib.rs (hunk 4 above)
Comment (gui1117) [reply substrate/frame/support/src/lib.rs]: maybe when we call in this:
```
		#[frame_support::register_default_impl(SolochainDefaultConfig)]
```
the registration is done in both std and no std. And it uses macro magic in the expansion somewhere

**Review by bkchr — APPROVED** [review]

**Review by koute — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Pull request merged**